scripts/copy_frag_an.py

```diff
@@ -35,51 +35,98 @@
+        if outart.type == 'ResultFile' and outart.name == 'Smear Analysis Result File':
+            try:
+                fid = outart.files[0].id
+                content['Smear_Analysis_Result_File'] = lims.get_file_contents(id=fid)
+            except:
+                log.append('No Smear Analysis Result File found')
+    #give error when there is no input file
+    if len(content)==0:
+        raise(RuntimeError("Cannot access any output file."))
     return content
 
 def get_data(csv_content, log):
     read=False
     data={}
-    text = csv_content.encode("utf-8")
-    # Try to determine the format of the csv:
-    dialect = csv.Sniffer().sniff(text)
-    pf = csv.reader(text.splitlines(), dialect=dialect)
-    #defaults
-    sample_index=1
-    conc_index=2
-    rin_index=3
-    ratio_index=4
-    for row in pf:
-        if 'Sample ID' in row:
-            #this is the header row
-            sample_index=row.index('Sample ID')
-            conc_index=row.index('Conc. (ng/ul)')
-            rin_index=row.index('RQN')
-            ratio_index=row.index('28S/18S')
-            read=True
-        elif(read and row[sample_index]):
-            #this is every other row
-            if row[sample_index] in data:
-                #Sample is duplicated, drop the key
-                log.append("sample {0} has two rows in the CSV file. Please check the file manually.".format(row[sample_index]))
-                del data[row[sample_index]]
-            else:
-                #normal procedure
-                data[row[sample_index]]={}
-                data[row[sample_index]]['concentration']=row[conc_index]
-                data[row[sample_index]]['rin']=row[rin_index]
-                data[row[sample_index]]['ratio']=row[ratio_index]
+    #read fragment analyzer result file
+    if csv_content.get('Fragment_Analyzer_Result_File'):
+        text = csv_content['Fragment_Analyzer_Result_File'].encode("utf-8")
+        # Try to determine the format of the csv:
+        dialect = csv.Sniffer().sniff(text)
+        pf = csv.reader(text.splitlines(), dialect=dialect)
+        #defaults
+        sample_index=1
+        conc_index=2
+        rin_index=3
+        ratio_index=4
+        sample_list=[]
+        for row in pf:
+            if 'Sample ID' in row:
+                #this is the header row
+                sample_index=row.index('Sample ID')
+                conc_index=row.index('Conc. (ng/ul)')
+                rin_index=row.index('RQN')
+                ratio_index=row.index('28S/18S')
+                read=True
+            elif(read and row[sample_index]):
+                if row[sample_index] not in sample_list:
+                    sample_list.append(row[sample_index])
+                    data[row[sample_index]]={}
+                    data[row[sample_index]]['concentration']=row[conc_index]
+                    data[row[sample_index]]['rin']=row[rin_index]
+                    data[row[sample_index]]['ratio']=row[ratio_index]
```

Comment:
Would it be good to try/except for IndexError in case the expected fields are missing?